simulator/util/sharedstore/sharedstore.go

```diff
@@ -0,0 +1,93 @@
+package sharedstore
+
+import (
+	"context"
+
+	"golang.org/x/xerrors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/util"
+)
+
+type Store interface {
```

Comment:
This is an important feature of this PR.
This SharedStore manages any `ResultStore`.